### DISCUSSION TITLE
Diagnostics for javascript

### DISCUSSION THREAD

**Discussion by Nuthin2do** [CodinGame/monaco-vscode-api#538] [Q&A] [answered]
Hey there,

i have a question regarding the diagnostics:

In my application i set up some some language defaults like this:

```
const typescriptDefaults = languages.typescript.typescriptDefaults;
    typescriptDefaults.setDiagnosticsOptions({
      noSemanticValidation: false,
      noSyntaxValidation: false,
    });
    const tsOp = typescriptDefaults.getCompilerOptions();
    tsOp.allowJs = true;
    tsOp.checkJs = true;
    tsOp.target = languages.typescript.ScriptTarget.ES2020;
    tsOp.lib = ["ES6"];
    tsOp.isolatedModules = true;
    typescriptDefaults.setCompilerOptions(tsOp);
```
    
Earlier, when i used the plain [monaco-api](https://github.com/microsoft/monaco-editor) i got red underlines when i wrote something undefined, for example like this (the setting up of my own *.d.ts-files works flawlessly):

<img width="819" alt="image" src="https://github.com/user-attachments/assets/dd922cac-706b-4e90-8d18-399e5b734fc0">

How can i achieve this with the monaco-vscode-api, too? Do i have to install a LSP for javascript (as described in the Wiki for python)? In the demo there is an error explicitly marked for this:

`function inc () {
  variable++
}`

in the main.common.ts, but if i write `console.ggg()` it is not marked as an error. Can anybody help me on that?

Thank you very much.

**Comment by CGNonofr** [comment]
There is 2 way to provide javascript/typescript intellisense

When you were using the plain monaco-editor, you were using the monaco-worker

It can be used with monaco-vscode-api by installing and importing `@codingame/monaco-vscode-standalone-typescript-language-features` and registering the worker `@codingame/monaco-vscode-standalone-typescript-language-features/worker` as 'typescript' and 'javascript'

**Reply by Nuthin2do** [reply]
[NewEditor.zip](https://github.com/user-attachments/files/17869133/NewEditor.zip)

I already did that... (see in the attached repo).

Thank very much for your help.

**Reply by CGNonofr** [reply]
I'm not sure to understand the issue then

**Reply by Nuthin2do** [reply]
I want to have the red underlining when i type something undefined. For example if i write in the repo i provided something like `console.ttt()` i don't get notified that "`ttt`" does not exist on `console` like any IDE would do that. If i write `console.ttt()` in VSCode (the appllication, not the API) for example, the `ttt()` is red underlined and on hovering over it i get a message like this ->

<img width="412" alt="image" src="https://github.com/user-attachments/assets/f5036986-2dd3-42ea-9f92-c1f683c34527">

And this red underlining with the notification i would like to have in the repo, too.

**Reply by Nuthin2do** [reply, marked as answer]
I found my mistake. Previously i set up the settings for typescript not for javascript. After changing that everything works fine:

`private setTypescriptValidationOptions() {
    const javascriptDefaults = languages.typescript.javascriptDefaults;
    javascriptDefaults.setDiagnosticsOptions({
      noSemanticValidation: false,
      noSyntaxValidation: false,
    });
    const tsOp = javascriptDefaults.getCompilerOptions();
    tsOp.allowJs = true;
    tsOp.checkJs = true;
    tsOp.target = languages.typescript.ScriptTarget.ES2020;
    tsOp.lib = ["ES6"];
    tsOp.isolatedModules = true;
    javascriptDefaults.setCompilerOptions(tsOp);
  }`
